documentation/add/04_mpi_framework.md

```diff
@@ -1,71 +1,108 @@
 
-The File-based framework has been written with the assumption that *job chunks may be processed independently*, but limited number of interprocess communications is occurring. This restricts the use of file-based framework to the correspondent part of Horace algorithms and for initial job submission. Matlab MPI and standard MPI frameworks do not have such restrictions but can not be used for job submission to the cluster.
+The File-based framework has been written with the assumption that *job chunks may be processed independently*, 
+but limited number of interprocess communications is occurring. 
+This restricts the use of file-based framework to the correspondent part of Horace algorithms and for initial job submission. 
+Matlab MPI and standard MPI frameworks do not have such restrictions but can not be used for job submission to the cluster.
 
-
-A cluster, executing Horace job using the wrappers should have the simple topology: 
+A cluster, executing a Horace job using the wrappers should have the following simple topology: 
 
 ![Fig 1: Horace Cluster](../diagrams/HoraceMPICluster.png)
 
 **Fig 1:** Cluster running Horace job with location of main software components of the parallel framework.
 
-The picture shows the interaction between main hardware components, necessary for Horace jobs and main software components, used to run these jobs. Black boxes on the **Fig 1** indicate the hardware components, namely parallel processes or parallel programs running Horace software, where the green boxes refer to the software  components, running on the appropriate hardware. The description of the software components is provided below. The red arrows on the picture refer to the message transfer media, used for communication between the processes/nodes. Currently implemented Horace jobs mainly communicate to/with the *headnode*, though its not a constrain but the details of current implementation. The frameworks allow efficient communication between nodes without any constrains. The Blue lines on the picture refer to the File-based message transfer, used to submit initial job and return job progress and information about the job results. There are also black arrows, referring to the process of the software submission to the cluster, described in more details in the chapter on **Common initialization**.
+The diagram shows the interaction between the main hardware and software components, used to run these jobs. 
+Black boxes in **Fig 1** indicate the hardware components, namely parallel processes or parallel programs running Horace software. 
+Green boxes refer to the software components, running on the appropriate hardware. 
+The red arrows on the picture refer to the message transfer medium, used for communication between the processes/nodes. 
+Currently implemented Horace jobs mainly communicate with the *head node*, though this is not a constraint, but the an implementation detail. 
```

Comment:
```suggestion
Currently implemented Horace jobs mainly communicate with the *head node*, though this is an implementation detail rather than a constraint.
```